src/transformers/models/propainter/modeling_propainter.py

```diff
@@ -0,0 +1,4583 @@
+        ds_mask_in_local = ds_mask_in[:, :local_timestep]
+        ds_mask_updated_local = F.interpolate(
+            masks_updated[:, :local_timestep].reshape(-1, 1, original_height, original_width),
+            scale_factor=1 / 4,
+            mode="nearest",
+        ).view(batch_size, local_timestep, 1, height, width)
+
+        if self.training:
+            mask_pool_l = self.max_pool(ds_mask_in.view(-1, 1, height, width))
+            mask_pool_l = mask_pool_l.view(batch_size, timestep, 1, mask_pool_l.size(-2), mask_pool_l.size(-1))
+        else:
+            mask_pool_l = self.max_pool(ds_mask_in_local.view(-1, 1, height, width))
+            mask_pool_l = mask_pool_l.view(
+                batch_size,
+                local_timestep,
+                1,
+                mask_pool_l.size(-2),
+                mask_pool_l.size(-1),
+            )
+
+        prop_mask_in = torch.cat([ds_mask_in_local, ds_mask_updated_local], dim=2)
+        _, _, local_features, _ = self.feature_propagation_module(
+            local_features, ds_flows_f, ds_flows_b, prop_mask_in, interpolation
+        )
+        encoder_hidden_states = torch.cat((local_features, ref_features), dim=1)
+
+        trans_feat = self.soft_split(
+            encoder_hidden_states.view(-1, num_channels, height, width),
+            batch_size,
+            fold_feat_size,
+        )
+        mask_pool_l = mask_pool_l.permute(0, 1, 3, 4, 2).contiguous()
+        trans_feat, all_hidden_states, all_self_attentions = self.transformers(
+            trans_feat,
+            fold_feat_size,
+            mask_pool_l,
+            t_dilation=t_dilation,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+        )
+
+        trans_feat = self.soft_comp(trans_feat, timestep, fold_feat_size)
+        trans_feat = trans_feat.view(batch_size, timestep, -1, height, width)
+
+        encoder_hidden_states = encoder_hidden_states + trans_feat
+
+        if self.training:
+            output = self.decoder(encoder_hidden_states.view(-1, num_channels, height, width))
+            output = torch.tanh(output).view(batch_size, timestep, 3, original_height, original_width)
+        else:
+            output = self.decoder(encoder_hidden_states[:, :local_timestep].view(-1, num_channels, height, width))
+            output = torch.tanh(output).view(batch_size, local_timestep, 3, original_height, original_width)
+
+        if not return_dict:
+            return tuple(v for v in [output, all_hidden_states, all_self_attentions] if v is not None)
+
+        return BaseModelOutput(
+            last_hidden_state=output,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+        )
+
+
+class ProPainterSpectralNorm(object):
+    # Invariant before and after each forward call:
+    #   u = normalize(W @ v)
+    # NB: At initialization, this invariant is not enforced
+
+    _version = 1
```

Comment:
What does this version entail?

Reply:
In the original code it was to maintain a record of changes made to the spectral normalization, to help ensure that state dictionaries created with older versions of the class can still be interpreted correctly. But as the class in here will remain unchanged so removing this part won't have any effects